docs/CLIUsage.md

```diff
@@ -1,23 +1,120 @@
+| repo-version | | | A specific repo version on the official GitHub repo; defaults to the most recent |
+| all-dependencies |  | true | Installs all dependency frameworks (with frameworkversions) it can find from the dependency section in `frameworkversions`|
+| auto-approve | | false | Skip interactive approval when existing version found. To always install the latest framework/frameworkversion |  
+| githubcredential | | | The file path to github credential file; defaults to $HOME/.git-credentials |
+| instance | | Framework name | The instance name. (default to Framework name) |
+| namespace | | "default" | The namespace where the operator watches for changes |
+
+### Environment Variables
+
+Environment Variables override flags and are intended to give the user more customizable CLI options.
+
+|  Environment Variable | Description  | 
+|---|---|
+| `GIT_USER`  |  Set a GitHub user to connect via the GitHub API with |
+| `GIT_PASSWORD` | Set a GitHub password to connect via the GitHub API with | 
+
+### Examples
+
+#### Install a Framework
+
+There are two options installing a Framework. While `kubectl apply -f *.yaml` is encouraged when developing and
+having `framework.yaml` or `frameworkversion.yaml` locally present, it is recommended to use the official Frameworks 
+provided in our [kubebuilder/frameworks](https://github.com/kudobuilder/frameworks) repo. `Kudoctl` offers a convenient
+way of installing those files in the command line.
+
+##### Install just the Framework without dependencies
 
-- Kudoctl
-- Kubectl Kudo Plugin
+This is the default behavior and just installs the according Framework.
 
-## Kudoctl
+```bash
+$ kubectl kudo install kafka
+framework.kudo.k8s.io/v1alpha1/kafka created
+frameworkversion.kudo.k8s.io/v1alpha1/kafka-2.11-2.4.0 created
+```
+
+#### Install a Framework with Dependencies
 
-The usage is intended to make your life easier when working with `Kudo`. A work flow would look like this:
+Sometimes you want to automatically install all dependencies a Framework comes with. This behavior is controlled via
+the flag `--all-dependencies`, and if set installs all dependencies a Framework has as well.
 
-1) You get a list of all available Instances deployed by `Kudo`
-2) You get the status the particular Instance of interest
-3) You see a history to a specific Instance
+```bash
+$ kubectl kudo install kafka --all-dependencies
+framework.kudo.k8s.io/v1alpha1/kafka created
+frameworkversion.kudo.k8s.io/v1alpha1/kafka-2.11-2.4.0 created
+framework.kudo.k8s.io/v1alpha1/zookeeper created
+frameworkversion.kudo.k8s.io/v1alpha1/zookeeper-1.0 created
+```
+
+#### Install a Framework with InstanceName & Parameters
+
+Use `--instance` and `--paramter`/`-p` for setting an Instance name or Parameter:
+
+```bash
+$ kubectl kudo install kafka --instance=my-kafka-name --parameter="KAFKA_ZOOKEEPER_URI,zk-zk-0.zk-hs:2181,zk-zk-1.zk-hs:2181,zk-zk-2.zk-hs:2181" --parameter="KAFKA_ZOOKEEPER_PATH,/small" -p="BROKERS_COUNTER,3"
```

Comment:
> This doesn't seem natural to me. There are two ways I could read `install`

I think we discussed this in our last community meeting on 3/7 which concluded that: A single holistic install command does not exclude having lower level maintenance commands for operators later on. 

> 1. Make the kafka framework available on the cluster to be used by application operators.
> 2. Run Kafka (e.g. `apt install mysql`) installs and runs MySQL on a computer
> 
> If this is supposed to be the first one, I wouldn't expect it to start up an instance during install
> If this is supposed to be the second one, I wouldn't expect the prompt

If no current instance is running the user is being asked if they want to continue creating the instance through a prompt `No Instance tied to this "kafka" version has been found. Do you want to create one? (Yes/no)`. When then simply saying `no` the cluster operator can opt out as it would mean that just the `Framework` and `FrameworkVersion` CRDs have been added/installed, no instantiation happened. But if he or she wants to save an extra step it also gives you this prompt to opt in with instantiating your `Instance` CRD. I am in favor of this, as we can still have lower level maintenance commands for operators but not under the cost of a imo better user flow. If an operator wants to install just the `Framework` CRD we could e.g. add later on the lower level `kubectl kudo install framework <name> [flags]` command. As @gerred said: 
```
it's like having "kubectl run" versus applying all of the relevant manifests individually

kubectl run being much more opinionated.

so high level, opinionated command with flags.

low level commands for escape hatches.

"kubectl run" versus applying all of the relevant manifests individually
```
 
> What happens with dependencies here? Do those get created as well? 

Same flow, it will also ask you if you want to create the dependency `Instance` CRD and if not you can simply say no and opt out.

> Can I set parameter values for those?

This is actually this thought here: https://github.com/kudobuilder/kudo/blob/51ffda81198fc67e16588506234bb37a907de50b/pkg/kudoctl/cmd/install/install.go#L249-L253 that would also introduce global parameters that will be applied if present to all dependency `Instance` CRDs.